tests/ui/self/elision/ref-assoc-async.rs

```diff
@@ -18,22 +17,27 @@ impl Trait for Struct {
 impl Struct {
     async fn ref_AssocType(self: &<Struct as Trait>::AssocType, f: &u32) -> &u32 {
```

Comment:
this now infers from `self` because the associated type constains the self type :sweat_smile: ah well, I guess it's fine